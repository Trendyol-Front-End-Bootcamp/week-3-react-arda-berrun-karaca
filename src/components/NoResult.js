import React from 'react';
import Button from './Button';

import './styles/NoResult.css';

function NoResult({ buttonClick }) {
  return (
    <div className="no-result">
      <img src={`${process.env.PUBLIC_URL}/assets/rick.png`} alt="Rick" />
      <h1>No Result</h1>
      <Button onClick={buttonClick}>Go Back</Button>
    </div>
  );
}

export default NoResult;
