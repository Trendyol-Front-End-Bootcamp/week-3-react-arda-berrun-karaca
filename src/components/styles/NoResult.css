.no-result {
    width: 90%;
    max-width: 600px;
    margin: 5rem auto 2rem auto;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    text-align: center;
    gap: 2rem;
}

.no-result img{
    max-width: 400px;
    height: auto;
    margin-left: 2rem;
}

.no-result h1{
    font-size: 2.6rem;
    color: var(--light-green); 
}

