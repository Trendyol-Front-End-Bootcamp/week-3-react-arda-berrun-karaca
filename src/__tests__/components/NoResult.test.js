import React from 'react';
import renderer from 'react-test-renderer';

import NoResult from '../../components/NoResult';

describe('NoResult tests', () => {
  it('should be defined', () => {
    expect(NoResult).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = renderer.create(<NoResult buttonClick={() => {}} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});