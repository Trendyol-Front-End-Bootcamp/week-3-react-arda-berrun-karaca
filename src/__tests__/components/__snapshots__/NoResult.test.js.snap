// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResult tests should render correctly 1`] = `
<div
  className="no-result"
>
  <img
    alt="Rick"
    src="/assets/rick.png"
  />
  <h1>
    No Result
  </h1>
  <button
    className="btn"
    onClick={[Function]}
  >
    Go Back
  </button>
</div>
`;
